File: Assets/Scripts/Core/FModuleInterface.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public enum F_MODULE_STATE
{
    /// <summary>
    /// 准备运行
    /// </summary>
    PREPARE,
    /// <summary>
    /// 运行中
    /// </summary>
    RUNNING,
    /// <summary>
    /// 暂停运行中
    /// </summary>
    PAUSE,
    /// <summary>
    /// 运行结束
    /// </summary>
    ENDED
}
public interface FModuleInterface
{
    F_MODULE_STATE STATE { get; set; }

    IEnumerator OnPrepare();

    int RelativeOrder { get; }

    void OnInitialize ();

    void OnRefresh();

    void OnPause();

    void OnResume();

    void Restart();
}

File: Assets/.lpproj/Apis/Unity/CS/FModuleInterface.lua
---@class CS.FModuleInterface
CS.FModuleInterface = {}

---@property readwrite CS.FModuleInterface.STATE : CS.F_MODULE_STATE
CS.FModuleInterface.STATE = nil

---@property readonly CS.FModuleInterface.Name : CS.System.String
CS.FModuleInterface.Name = nil

---@return CS.System.Collections.IEnumerator
function CS.FModuleInterface:OnPrepare()
end

function CS.FModuleInterface:OnInitialize()
end

function CS.FModuleInterface:OnRefresh()
end

function CS.FModuleInterface:OnPause()
end

function CS.FModuleInterface:OnResume()
end

function CS.FModuleInterface:Restart()
end